modules/allpass.h

```diff
@@ -0,0 +1,68 @@
+// #Allpass
+// Allpass filter module
+// ## Credits
+// *Original author(s) : Barry Vercoe, John ffitch
+// *Year : 1991
+// *Ported from soundpipe by Ben Sergentanis, May 2020
+
+#pragma once
+#ifndef DSY_ALLPASS_H
+#define DSY_ALLPASS_H
+
+#include <stdint.h>
+#ifdef __cplusplus
+
+#include <math.h>
+
+namespace daisysp
+{
+
+class Allpass
+{
+    public:
+    Allpass() {}
+    ~Allpass() {}
+  
+    // ### Init
+    //
+    // Initializes the allpass module.
+    //
+    // sample_rate - The sample rate of the audio engine being run. 
+    // 
+    // ~~~~
+        void Init(float sample_rate);
+    // ~~~~
+
+    // ### Process
+    //
+    // 
+    // ~~~~
+        float Process(float in);
+    // ~~~~
+
+    // #Setters
+    //
+    // ### SetLoopTime
+    //
+    // 
+    // ~~~~
+        inline void SetLoopTime(float looptime) { loop_time_ = fminf (0.5f, looptime); buf_size_ = (int) (loop_time_ * sample_rate_); }
+    // ~~~~
+
+    // #Setters
+    //
+    // ### SetRevTime
+    //
+    // ~~~~
+        inline void SetRevTime(float revtime) { rev_time_ = revtime; }
+    // ~~~~
+	
+    
+    private:
+	float sample_rate_, rev_time_, loop_time_, 
+	  prvt_, coef_, buf_[96000];
```

Comment:
So, actually with the large buffer needed we may actually want to make this a template (like DelayLine). So that the size can be set at compile time.

The alternative would be to have an externally supplied buffer (i.e. `float *buf` with a size parameter. Both of which would be passed in at init). 

Making it a template, would also mean inlining the entire header file, and with math.h usage that gets a little trickier.